apache2/waf_logging/waf_log_util.cc

```diff
@@ -208,7 +209,8 @@ char* generate_json(const char* resource_id, const char* operation_name, const c
         const char* instance_id, const char* client_ip, const char* client_port, const char* request_uri,
         const char* rule_set_type, const char* rule_set_version, const char* rule_id, const char* messages,
         const int action, const int site, const char* details_messages, const char* details_data,
-        const char* details_file, const char* details_line, const char* hostname, const char* waf_unique_id) {
+        const char* details_file, const char* details_line, const char* hostname, const char* waf_unique_id, 
+        const char* timestamp) {
```

Comment:
It looks like we are passing the `timestamp` in as a parameter but never use it inside the `generate_json` function. I think we should propagate it further to the call to `get_json_log_message` call and populate the serialized structure with it.

Reply:
besides test on the local server, it's better to do the E2E testing on the gateway so you could see the jarvis log.